### ISSUE TITLE
FTP passive mode fails due to SO_REUSEPORT/SO_REUSEADDR is not set

### ISSUE THREAD

**Issue by perillamint** [bolcom/libunftp#526]
When the client opens and closes the PASV FTP connection fast enough, libunftpd will fail to transfer the file with the following error

```
425 No data connection established
```

This is caused because `tokio::net::TcpSocket::bind()` does not set `SO_REUSEADDR` and `SO_REUSEPORT`, thus disrupting binding address/port reuse.

This can be solved using the following code in `Pasv::try_port_range`, but due to the `unsafe` nature of `libc`, `unsafe_code = "deny"` restriction should be released.

```
            let s = TcpSocket::new_v4()?;
            unsafe {
                let optval: libc::c_int = 1;
                let ret = libc::setsockopt(
                    s.as_raw_fd(),
                    libc::SOL_SOCKET,
                    libc::SO_REUSEADDR,
                    &optval as *const _ as *const libc::c_void,
                    mem::size_of_val(&optval) as libc::socklen_t,
                );
                if ret != 0 {
                    return Err(io::Error::last_os_error());
                }
                let ret = libc::setsockopt(
                    s.as_raw_fd(),
                    libc::SOL_SOCKET,
                    libc::SO_REUSEPORT,
                    &optval as *const _ as *const libc::c_void,
                    mem::size_of_val(&optval) as libc::socklen_t,
                );
                if ret != 0 {
                    return Err(io::Error::last_os_error());
                }
            }
```

**Comment by robklg** [comment]
Thanks for catching this and reporting it. Apparently this problem exists since v0.20.0, where we switched from TcpListener to TcpSocket. That apparently caused that the reuseaddr option is no longer automatically enabled, and also not specifically disabled.

**Comment by robklg** [comment]
Fix is released in v0.20.3 and unFTP v0.15.1
